lib/saml_idp/request.rb

```diff
@@ -171,7 +171,7 @@ def matching_cert
       Array(service_provider.certs).find do |cert|
         document.valid_signature?(
           OpenSSL::Digest::SHA256.new(cert.to_der).hexdigest,
-          options.merge(cert:)
+          options.merge(cert:, no_namespace_enabled: true)
```

Comment:
just to clarify, `no_namespace_enabled: true` is the same as old legacy behavior, right?

Reply:
it's confusing because not using a namespace was technically allowed in the old behavior, but because of the bug this one node that couldn't be located and it didn't work correctly. (i'm struggling to find a terse name for this.)

Reply:
actually i have an idea (and maybe can restructure a bit to make this more testable)

Reply:
i upated it to `digest_method_fix_enabled` to make it more clear and moved the conditional logic into the method so it was more testable.

https://github.com/18F/saml_idp/pull/104/commits/5ad9e188efdfa6597697dd87f9cb9e8efa8d7d09